extensions/legit/legit.lisp

```diff
@@ -538,11 +545,74 @@ Currently Git-only. Concretely, this calls Git with the -w option.")
                            ;; Save the hash on this line for later use.
                            (when hash
                              (put-text-property start point :commit-hash hash)))))
-              (lem/peek-legit:collector-insert "<none>")))
+              (lem/peek-legit:collector-insert collector "<none>")))
 
         (add-hook (variable-value 'after-change-functions :buffer (lem/peek-legit:collector-buffer collector))
                   'change-grep-buffer)))))
 
+(defun get-clgit-walker (path)
+  (arrow-macros:-> path
+    lem-core/commands/project:find-root
+    cl-git:open-repository
+    cl-git:repository-head
+    cl-git:target
+    (lambda (x) 
+      (cl-git:revision-walk x :ordering :topological))))
+
+(define-command legit-log () ()
+  "Show log buffer."
+  (with-current-project ()
+    (lem/peek-legit:with-collecting-sources (collector :read-only nil)
+      (if (not (lem/porcelain:git-project-p))
```

Comment:
so until now we did that check and dispatch on porcelain.lisp. See `porcelain-error`.